src/storagemanager/src/main/java/org/vpac/ndg/task/ITask.java

```diff
@@ -48,5 +48,8 @@
 	 * @return Returns the description of the task.
 	 */
 	public String getDescription();
-
+	/**
+	 * Get the contribution of this task to task piplines overall progress
+	 */
+	public double getProgressWeight();

```

Comment:
Java 8.0 now allows default implementations of functions in an interface. So you could return 1.0 here and override it for certain tasks.